### PR TITLE
Reapply gogs hmac patch

### PR DESCRIPTION
**Pull request by zobo**
Do not send "secret" field in webhook json but rather a header with HMAC.

Sending X-Gogs-Signature header and X-Hub-Signature with sha265= prefix.

https://github.com/gogs/gogs/commit/3609efe2e3d562547ae6497d67d56be7626727d0

https://github.com/go-gitea/gitea/issues/3901

### CODE REVIEW

**Comment by lafriks** [comment]
Please send PR to SDK first

**Comment by zobo** [comment]
> Please send PR to SDK first

Thanks, will do.

File: models/webhook.go [hunk 1]
```diff
@@ -731,6 +730,8 @@ func (t *HookTask) deliver() {
 		Header("X-Gitea-Event", string(t.EventType)).
 		Header("X-Gitea-Signature", t.Signature).
 		Header("X-Gogs-Delivery", t.UUID).
+		Header("X-Gogs-Signature", t.Signature).
```
Comment (techknowlogick) [inline models/webhook.go:733]: Gogs signature is already set two lines down

**Review by techknowlogick — CHANGES_REQUESTED** [review]

**Review by techknowlogick — CHANGES_REQUESTED** [review]
This breaks Drone integration. Drone should get sent a patch to handle either the secret field OR validate the header, that way Drone is backwards compatible with older versions of Gitea too.

**Comment by zobo** [comment]
> This breaks Drone integration. Drone should get sent a patch to handle either the secret field OR validate the header, that way Drone is backwards compatible with older versions of Gitea too.

Agreed, will work on that too.

**Pull request closed without merging**